### ISSUE TITLE
coordd/dataflowd: Trouble sending large commands

### ISSUE THREAD

**Issue by antiguru** [MaterializeInc/materialize#9084]
From #8947: Processing a large-ish join causes the dataflowd-s to exit with ```dataflowd: frame size too big```. It seems that the "too big" in question is 67Mb, which is not that much in the grand scheme of things. Please let me know if a test case is required for this, I have it repeatable.

**Comment by philip-stoev** [comment]
@benesch may have sorted that out already in hia follow up PR . I recall
one pf his commits messages was to that effect

**Comment by antiguru** [comment]
That's correct, but the solution presented in https://github.com/MaterializeInc/materialize/pull/9062/commits/4f8fee4bbb830e7682529f6dd0adf5e278da0194 is not final. Currently it can be tricked into retaining large allocations forever, which we'd like to avoid in the future.
